nova/virt/vmwareapi/vmops.py

```diff
@@ -1497,19 +1498,16 @@ def _destroy_instance(self, context, instance, destroy_disks=True):
                 cluster = cluster_util.fetch_cluster_properties(self._session,
                                                                        vm_ref)
                 config_info_ex = cluster.propSet[0].val
-
                 if hasattr(config_info_ex, 'group'):
-                    for key, group in enumerate(config_info_ex.group):
+                    for group in config_info_ex.group:
                         if not hasattr(group, 'vm'):
                             continue
-
-                        for vm in group.vm:
-                            if vm.value == vm_ref.value and len(group.vm) == 1:
-                                cluster_util.delete_vm_group(
-                                            self._session, cluster.obj,
-                                            config_info_ex.group[key])
-                                break
-                        break
+                        else:
+                            for vm in group.vm:
+                                if vm.value == vm_ref.value:
+                                    vm_group = group
+                                    break
```

Comment:
Couldn't there be multiple groups in theory? If an admin added a VM to a group via vCenter or something?